server/src/main/java/io/crate/interval/NumericalIntervalParser.java

```diff
@@ -97,8 +97,16 @@ private static Period roundToPrecision(int value,
                 return Period.seconds(value);
             }
         }
-        if (start == IntervalParser.Precision.SECOND && end == null) {
-            return buildSecondsWithMillisPeriod(value, millis);
+        if (start == IntervalParser.Precision.SECOND) {
+            if (end == null) {
+                return buildSecondsWithMillisPeriod(value, millis);
+            }
+            if (end == IntervalParser.Precision.MILLISECOND) {
+                return buildSecondsWithMillisPeriod(0, value * 1000 + millis);
+            }
+        }
+        if (start == IntervalParser.Precision.MILLISECOND && end == null) {
+            return buildSecondsWithMillisPeriod(0, value);
```

Comment:
Is it possible scenario to have MILLISECOND precision as start?  
See also `testIntervalInvalidStartEndMillis`

Reply:
The interval in Postgres does not support MILLISECOND as start or end, so I removed it.